Remove 1.57 deprecation warnings

OmarShehata opened the pull request:
This removes the deprecation warnings for fetchImage https://github.com/AnalyticalGraphicsInc/cesium/issues/7630 and followSurface https://github.com/AnalyticalGraphicsInc/cesium/issues/7504.

@tfili

cesium-concierge commented:
Thanks for the pull request @OmarShehata!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/deprecation-warnings/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

tfili commented:
@OmarShehata You have 2 failing tests.

Also, thanks for getting to these.

OmarShehata commented:
@tfili I ended up having to not just remove `followSurface` from the public API, but also all uses internally, because otherwise, the constructor no longer accepted the option, but you could then set it on a polyline directly and it would work. 

So I removed it, and updated the tests. Two tests were completely removed:

* `PolylineGeometrySpec.js / constructor converts followSurface to arcType`, because it was testing whether the constructor correctly converts this deprecated option.
* `PolylineGeometryUpdaterSpec.js  / followSurface true with undefined globe does not call generateCartesianArc`, because it was just a duplicate of the test below it for followSurface instead of arcType.

I did NOT remove anything in the CZML adapting the option because that's intentionally there for backwards compatibility. The polyline Sandcastles seem to work correctly.
